### PR TITLE
Update to PHP 7.2

### PR DESCRIPTION
**Pull request by thomwiggers**
Also includes better way of handling packages

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,35 @@
-FROM php:7.1-fpm
+FROM php:7.2-fpm
 
-# Based on https://github.com/docker-library/wordpress/blob/d122aa204948aecd651af435fe4ca058bb99545d/php7.1/fpm/Dockerfile
+# Based on https://github.com/docker-library/wordpress/blob/master/php7.2/fpm/Dockerfile
```
Comment (pingiun) [inline Dockerfile]: Linking to master seems worse than the actual commit hash, why did you choose to do this?

**Review by joren485 — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
 	\
 	docker-php-ext-configure gd --with-png-dir=/usr --with-jpeg-dir=/usr; \
-	docker-php-ext-install gd mysqli opcache
-# TODO consider removing the *-dev deps and only keeping the necessary lib* packages
+	docker-php-ext-install gd mysqli opcache zip; \
+	\
+# reset apt-mark's "manual" list so that "purge --auto-remove" will remove all build dependencies
+	apt-mark auto '.*' > /dev/null; \
+	apt-mark manual $savedAptMark; \
+	ldd "$(php -r 'echo ini_get("extension_dir");')"/*.so \
```
Comment (pingiun) [inline Dockerfile:22]: This seems like a hacky way to do this. Also this is not very readable without taking the time to look at it.

Comment (thomwiggers) [reply Dockerfile:22]: I stole it from the Wordpress repo

File: Dockerfile (hunk 1 above)
Comment (thomwiggers) [reply Dockerfile]: Laziness.

**Pull request merged**